airflow/config_templates/unit_tests.cfg

```diff
@@ -59,6 +59,8 @@ unit_test_mode = True
 killed_task_cleanup_time = 5
 # We only allow our own classes to be deserialized in tests
 allowed_deserialization_classes = airflow\..* tests\..*
+# mocks from unittest.mock work correctly in subprocesses only if they are created by "fork" method
+mp_start_method = fork
```

Comment:
It's know issue that MacOS doesn't work correctly in fork mode in Python (https://github.com/python/cpython/issues/58037 and https://bugs.python.org/issue30385#msg293958), that basically why spawn default method in python. 

Airflow itself use `fork` in some places [even in MacOS](https://github.com/apache/airflow/blob/b82ce61285f3f4f0c7eccb2c3effaef53c9fb84e/airflow/settings.py#L552) and that the main reason for Segmentation Faults, in some other places [it is controlled by](https://github.com/apache/airflow/blob/58d8577f3528e7e143addefb03711bdc82cb1782/airflow/utils/mixins.py#L41-L42) `mp_start_method`.

So I would be very surprised if this change fix in one place and do not break in another. In conclusion I want to say that some tests just not work in MacOS and developer should use Breeze however I have no objection to adding this.

Reply:
If this change only for internal airflow tests might be better add it into `conftest.py`

https://github.com/apache/airflow/blob/7700fb12cc6c7a97901662e6ac6aa1e4e932d969/tests/conftest.py#L38-L44

Reply:
I understand. I knew of the problem with fork in macOS but I thought it's just unstable but functionally correct for Airflow purposes, so I figured it's still useful to switch it for testing. I will review other tests to see if there is any regression on macOS after this change. I personally use breeze but it's a little cumbersome to use when debugging code with IDE.

Reply:
What is the difference between adding to unit_tests.cfg and conftest.py?

Reply:
Use or not `unit_tests.cfg` controlled by [`AIRFLOW__CORE__UNIT_TEST_MODE`](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#unit-test-mode) and will affect all Airflow users who use this variable in their tests

In opposite `conftest.py` only for internal Airflow Tests, and only affect contributors to Airflow, who might run tests in Breeze. In additional some tests could override `AIRFLOW__CORE__UNIT_TEST_MODE` to False.

That is my thoughts, lets wait what he other says. Maybe it could really be better place it in `unit_tests.cfg`

Reply:
> What is the difference between adding to unit_tests.cfg and conftest.py?

To add what @Taragolis wrote - this has also changed recently - it used to be independently set but as of last month or so I changed it in the way that `AIRFLOW__CORE__UNIT_TEST_MODE` is actually set in `conftest.py` - so basically all Airflow unit tests always have the `AIRFLOW__CORE__UNIT_TEST_MODE` set to True. This was all part of restructuring the configuration.

Right now - no matter how you run your tests they will have it set to True.

So that configuration can be contributed by providers. So it could be set in either of the two places. 

However, I think it's better to set this variable in conftest.py and only for MacOS because you can set it conditionally only when MacOS is detected. That's the nice thing about `conftest.py` that you can set variables and test configuraiton dynamically - by writing the imperative code rather than declare the configuration.

This might have the nice thing that eventually somoene could run the full test suite on MacOS as well and fix all the issues that introduces (also conditionally) - while CI would continue to run the default "fork" mode in CI.

Reply:
OK，I moved the change to `conftest.py` anded condition for MacOS only.
I looked at other tests and didn't notice any negative impact of the tests on my machine.